hypervisor/src/mshv/mod.rs

```diff
@@ -611,6 +611,43 @@ impl cpu::Vcpu for MshvVcpu {
                         info.interrupt_vector.try_into().unwrap(),
                     ))
                 }
+                hv_message_type_HVMSG_X64_SEV_VMGEXIT_INTERCEPT => {
```

Comment:
Should this be put behind the `sev_snp` feature gate?

Reply:
I don't think so because this exit cannot happen on a non sev-snp guest.

Reply:
If the exit is only for `sev-snp` guests, it should be put behind the `sev_snp` feature gate.

Reply:
I mean in future we would like to get rid of sev-snp feature gate right? Instead have a unified Cloud Hypervisor binary which works for all the scenario. All these code is already runtime dependent so why put this behind a feature gate.

Reply:
@likebreath what do you think?

Reply:
No, I have to disagree. Removing the `sev-snp` feature gate is a separate topic and will be handled at once when the time comes.

The purpose of having a feature gate is to define code dependency at compile time (statically). This is good for code maintenance, reducing dead-code, and avoid unexpected runtime errors (say errors from this code block when `sev-snp` featuer is not enabled).

Reply:
Ack. Updated with a feature gate.